srcCppCommon/PerftestPrinter.cxx

```diff
@@ -0,0 +1,483 @@
+        return LEGACY;
+    } else {
+        return CSV;
+    }
+}
+
+PerftestPrinter::PerftestPrinter()
+{
+    _dataLength = 100;
+    _outputFormat = CSV;
+    _printSummaryHeaders = true;
+}
+
+void PerftestPrinter::initialize(bool printIntervals,
+        std::string outputFormat,
+        bool printHeaders,
+        bool printSerialization,
+        bool showCpu)
+{
+    _printIntervals = printIntervals;
+    _headerPrinted = printHeaders;
+    _printSerialization = printSerialization;
+    _showCPU = showCpu;
+    if(outputFormat == "csv")
+        _outputFormat = CSV;
+    else if(outputFormat == "json"){
+        _outputFormat = JSON;
+        _controlJsonInit = true;
+    }else if(outputFormat == "legacy")
+        _outputFormat = LEGACY;
+}
+
+void PerftestPrinter::set_data_length(unsigned int dataLength)
+{
+    _dataLength = dataLength;
+}
+
+void PerftestPrinter::set_header_printed(bool headerPrinted)
+{
+    _headerPrinted = headerPrinted;
+}
+
+void PerftestPrinter::print_latency_interval_header()
+{
+    switch(_outputFormat){
+        case CSV:
+            if(_headerPrinted && _printIntervals){
+                printf("\nIntervals One-way Latency for %d Bytes:\n",
+                _dataLength);
+                printf("Length (Bytes)"
+                        ", Latency (" PERFT_TIME_UNIT
+                        "), Ave (" PERFT_TIME_UNIT
+                        "), Std (" PERFT_TIME_UNIT
+                        "), Min (" PERFT_TIME_UNIT
+                        "), Max (" PERFT_TIME_UNIT ")");
+                if(_showCPU)
+                    printf(", CPU %%");
+
+                printf("\n");
+            }
+            break;
+        case JSON:
+            if(_controlJsonInit){
+                _controlJsonInit = false;
+            } else {
+                printf(",\n\t\t{\n");
+            }
+            printf("\t\t\t\"length\":%d,\n",_dataLength);
+
+            if(_printIntervals && _outputFormat == JSON){
+                printf("\t\t\t\"intervals\":[\n");
+                _controlJsonIntervals = true;
+            }
+            break;
+        case LEGACY:
+            if(_headerPrinted && _printIntervals){
+                printf("\n\n********** New data length is %d\n",
+                        _dataLength);
+            }
+            break;
+    }
+}
+
+void PerftestPrinter::print_throughput_header()
+{
+    switch(_outputFormat){
+        case CSV:
+            if(_headerPrinted && _printIntervals){
+                printf("\nInterval Throughput for %d Bytes:\n",
+                _dataLength);
+                printf("Length (Bytes), Total Samples,  Samples/s,"
+                        " Ave Samples/s,     Mbps,  Ave Mbps"
+                        ", Lost Samples, Lost Samples (%%)");
+                if(_showCPU)
+                    printf(", CPU %%");
+
+                printf("\n");
+            }
+            break;
+        case JSON:
+            if(_controlJsonInit){
```

Comment:
I suggest we call this variable `_isJsonInitialized`

Reply:
Maybe the true/false in that case change the meaning, right? check that.

Reply:
ok